node/node.go

```diff
@@ -456,36 +432,40 @@ func NewNode(config *cfg.Config,
 	}
 
 	p2p.MultiplexTransportConnFilters(connFilters...)(transport)
+	return transport, peerFilters
+}
+
+func buildNewNodeSwitch(config *cfg.Config,
```

Comment:
```sugggestion
func createSwitch(config *cfg.Config,
```

Why `NewNode` prefix is necessary?

Reply:
Initially it was to make it clear where the functions were extracted from. I thought that perhaps if they were eventually used in tests that do partial node instantiations using those functions it'd also make it clearer that they were called by the `NewNode` function.

But I suppose that in itself would only be an intermediate step. The ultimate goal would be to have a simpler interface to do partial node instantiations (e.g. some kind of `NewNode` function with a whole bunch of flags to indicate which parts of the node to create, where this function would be used by tests that want partial node instantiations).